### ISSUE TITLE
Upgrade fails if an index has the name "value"

### ISSUE THREAD

**Issue by hgl** [indexeddbshim/IndexedDBShim#199]
``` js
req.onupgradeneeded = function () {
   var store = db.createObjectStore("ip");
   store.createIndex("value", "ip");

   store = db.createObjectStore(....);
};
```

This seems to cause all later object store creations to be ignored, and the solely created `ip` store have an empty `{}` for `indexList` column in `__sys__` table.

**Comment by hgl** [comment]
After some investigation, the problem is that i used `value` for the index name, and it collided with the existing column. Looks like you can't call an index `key` or `value`?

**Comment by mattjacobsen** [comment]
I had something similar with a store named "user" and an index named
"table". They're reserved keywords for webSQL and it mucks up the
statements IndexedDBShim uses to create things. I was going to submit a
patch for this, but didn't think it would affect enough (if any) users to
make it worthwhile.

Thinking about it, I guess if you wanted to, you could perform sql
injection attacks on IndexedDBShim/webSQL with the code at present. Hmm,
maybe I should have submitted that patch after all... ;-)

**Comment by hgl** [comment]
I think you should. :)

Is it still able to be merged? How do you workaround it? I'm not too familiar with web sql, but I wonder if this would work: keep a list of keywords, and whenever store/index name is one of them, prefix it with some constant string.

**Comment by open-environment** [comment]
encountered this issue too.

**Comment by brettz9** [comment]
The WebSQL keyword conflict can apparently be avoided by using backticks, but there is still the problem of conflict with our own internal tables.

For now, in `master`, I've avoided the index problems (conflicts with our `key`, `value`, and potentially `inc` columns), by prefixing all user index columns with `_`.

There could also have been a problem if a database had been named "__sysdb__.sqlite" (or began with the reserved `sqlite_`) or a store had been named "__sys__". I've fixed these in `master` by prefixing user databases/stores with `D_` and `s_`, respectively.

If it's really a problem for people, I could add logic to detect our internal columns (or tables//databases) as suggested by @hgl which would make the db names and such a little cleaner, but I think that is a little less secure in case down the road someone adds a new table, column, or database within IndexedDBShim.

I'll keep the issue open for a while to gather feedback on the approach, but it should be currently resolved.
